File: 0071-simplify-path/0071-simplify-path.go
func simplifyPath(path string) string {
 	var directoryStack []string

 	pathComponents := strings.Split(path, "/")

	for _, component := range pathComponents {
 		if component == "" || component == "." {
			continue
		}

 		if component == ".." {
 			if len(directoryStack) > 0 {
				directoryStack = directoryStack[:len(directoryStack)-1]
			}
		} else {
 			directoryStack = append(directoryStack, component)
		}
	}
 	return "/" + strings.Join(directoryStack, "/")
}

File: 0071-simplify-path/0071-simplify-path.js
/**
 * @param {string} path
 * @return {string}
 */
var simplifyPath = function(path) {
    const directoryStack = [];

    const components = path.split('/');

    for (const component of components) {
        if (component === '' || component === '.') { // Cur Dir
            continue;
        }

        if (component === '..') {
            if (directoryStack.length > 0) {
                directoryStack.pop();
            }
        } else {
            directoryStack.push(component);
        }
    }
    return '/' + directoryStack.join('/');
}